### PR TITLE
Support accessing to options in response hook

### LINKED ISSUE #114: Support accessing options in afterResponse hook
There is a case where I need to behave differently in afterResponse hook according to the parameters passed in options, so access options in afterResponse hoook is very useful.

It is an example, I extend ky, when error occurs, global messages can be display or not according to `silent` passed in `_ky.get()`.
```js
const _ky = ky.extend({
  hooks: {
    afterResponse: [
      (res, { silent }) => {
        if (!res.ok && !silent) {
          // display global messages...
        }
        return res
      }
    ]
  }
})

_ky.get('example.com/api/xx', { silent: false })  // if error occurs, global message will display
_ky.get('example.com/api/xx', { silent: true })   // if error occurs, global message will not display
```

### PR DESCRIPTION
**Pull request by kangkai124**
Fixes #114

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -209,7 +209,7 @@ class Ky {
 
 			for (const hook of this._hooks.afterResponse) {
 				// eslint-disable-next-line no-await-in-loop
-				const modifiedResponse = await hook(response.clone());
+				const modifiedResponse = await hook(response.clone(), this._options);
```
Comment (szmarczak) [inline index.js]: This should be documented.

**Comment by sholladay** [comment]
I'm okay with this idea. However, before we do it, we should think about which options people should see in the options object. You might, for example, expect the options object to contain `timeout` and `throwHttpErrors`, but it won't, as implemented by this PR. I'd say that's likely to confuse users, though obviously we can just document it that way. Thoughts?

Also see https://github.com/sindresorhus/ky/issues/100, where I explain a similar issue.

**Comment by kangkai124** [comment]
@sholladay The problem you pointed makes sense. May be we could tell people that options object in afterResponse is not all same with what they passed to ky. For example, we can not get `timeout` by default, and we pass `searchParams object` to ky, but get parsed `query string` in afterResponse. One solution is that we add those params, which useful in afterResponse, to a special object such as called `resOptions`, like this:
```
ky.get(url, {
  timeout: 2000,
  resOptions: {
    silent: false,
    duration: 800
  }
})
```
And if you agree, I am willing to write it to document.

File: index.js (hunk 1 above)
Comment (kangkai124) [reply index.js]: Readme has been updated.

**Comment by sholladay** [comment]
I've been thinking about this and I'm starting to lean towards having all options be passed to the hooks, so that we can have a simpler API and simpler TypeScript types. We can selectively remove properties before passing them to `fetch()` if it ever becomes necessary, but for now, it isn't.

The next problem is just making sure the documentation is clear about which options the hooks are capable of modifying

**Comment by szmarczak** [comment]
I agree with @sholladay

**Comment by sholladay** [comment]
I updated the code so the hooks now both have `input` and `options` as their first two arguments, just like Ky and `fetch()` have. The `afterResponse` hook has `response` as a third argument. So this is a breaking change.

We still need to update how the options are destructured so all of the options actually end up on that options object passed to the hooks. Will have to do that later.

**Comment by sindresorhus** [comment]
The TypeScript definition also needs to be updated.

**Comment by sindresorhus** [comment]
@kangkai124 Bump

**Comment by sindresorhus** [comment]
Closing in favor of https://github.com/sindresorhus/ky/pull/136.

**Pull request closed without merging**